cmd/agent/app/flags.go

```diff
@@ -24,54 +24,71 @@ import (
 	"flag"
 	"fmt"
 	"strings"
+	"github.com/spf13/viper"
 )
 
+type stringSliceFlag struct {
+	slice *[]string
+}
+
 // Bind binds the agent builder to command line options
-func (b *Builder) Bind(flags *flag.FlagSet) {
+func (b *Builder) GenerateFlags() *flag.FlagSet {
+	flagSet := &flag.FlagSet{}
+
 	for i := range b.Processors {
 		p := &b.Processors[i]
 		name := "processor." + string(p.Model) + "-" + string(p.Protocol) + "."
-		flags.IntVar(
-			&p.Workers,
+		flagSet.Int(
 			name+"workers",
 			p.Workers,
 			"how many workers the processor should run")
-		flags.IntVar(
-			&p.Server.QueueSize,
+		flagSet.Int(
 			name+"server-queue-size",
 			p.Server.QueueSize,
 			"length of the queue for the UDP server")
-		flags.IntVar(
-			&p.Server.MaxPacketSize,
+		flagSet.Int(
 			name+"server-max-packet-size",
 			p.Server.MaxPacketSize,
 			"max packet size for the UDP server")
-		flags.StringVar(
-			&p.Server.HostPort,
+		flagSet.String(
 			name+"server-host-port",
 			p.Server.HostPort,
 			"host:port for the UDP server")
 	}
-	flags.Var(
+	flagSet.Var(
 		&stringSliceFlag{slice: &b.CollectorHostPorts},
+		//stringSlice,
 		"collector.host-port",
 		"comma-separated string representing host:ports of a static list of collectors to connect to directly (e.g. when not using service discovery)")
-	flags.StringVar(
-		&b.SamplingServer.HostPort,
+	flagSet.String(
 		"http-server.host-port",
 		b.SamplingServer.HostPort,
 		"host:port of the http server (e.g. for /sampling point)")
-	flags.IntVar(
-		&b.DiscoveryMinPeers,
+	flagSet.Int(
 		"discovery.min-peers",
 		3,
 		"if using service discovery, the min number of connections to maintain to the backend")
+
+	return flagSet;
 }
 
-type stringSliceFlag struct {
-	slice *[]string
+func (b *Builder) Parse() {
+	for i := range b.Processors {
+		p := &b.Processors[i]
+		name := "processor." + string(p.Model) + "-" + string(p.Protocol) + "."
+		p.Workers = viper.GetInt(name+"workers");
```

Comment:
Can this work without global var? E.g. with standard `flags` you can create a FlagSet and use it without relying on global state.